packages/cozy-konnector-libs/src/libs/saveBills.js

```diff
@@ -197,4 +203,68 @@ function checkRequiredAttributes(entries) {
     }
   }
 }
+
+function appendContractsDatas(tempEntries, options) {
+  if (options.contractLabel && options.contractId === undefined) {
+    log('warn', 'contractLabel used without contractId')
+    return tempEntries
+  }
+  let newEntries = tempEntries
+  if (options.contractId) {
+    // if contractId passed by option
+    newEntries = appendContractDatasByOption(newEntries, options)
+  } else if (billsHaveContractId(newEntries)) {
+    // if contractId passed by bill attribute
+    newEntries = appendContractDatasByAttribute(newEntries, options)
+  }
+  return newEntries
+}
+
+function appendContractDatasByOption(newEntries, options) {
+  // Surcharge label if not define
+  if (!options.contractLabel) {
+    options.contractLabel = options.contractId
+  }
```

Comment:
options passed from parent is changed here. This is dangerous as it is unexpected from the parent. Please make a copy of options or use other variables.

Reply:
Comme discuté avec Eric, c'est fait en connaissance de cause.
On a pris le parti de crée deux copies (entries et options) en début de la fonction saveBills pour limiter toute erreur de la part d'un utilisateur du saveBills.

Mais pour le dev,  c'est fait sciamment tout au long de cette fonction, les objets entries et options sont édités suivant le context, puis utilisé. On va pas ré-écrire l'ensemble de la fonction.

Est ce que cette justification suffit à lever tes craintes sur cet objet option ?

Reply:
Non. Je ne considère pas qu'il y ait une grande différence entre l'utilisateur de saveBills et le dev qui va utiliser la fonction appendContractData dans 6 mois et qui aura zappé que l'objet `options` est modifié.

Ca n'est pas précisé dans la docstring de la fonction, c'est très facile à oublier et derrière ca fait des bugs difficiles à trouver parce qu'on ne sait pas qui modifie l'objet d'options.

Reply:
> Comme discuté avec Eric, c'est fait en connaissance de cause.

Comme discuté ou

Reply:
En audio.

Reply:
> Non. Je ne considère pas qu'il y ait une grande différence entre l'utilisateur de saveBills et le dev qui va utiliser la fonction appendContractData dans 6 mois et qui aura zappé que l'objet options est modifié.

`appendContractData` en me semble pas spécialement exposée tout de même.

Et, par principe, saveBills est une "meta fonction", faite pour définir les bonnes options passées à saveFiles et hydrateAndFilter dans les cadre des factures. Du coup, ça me semble approprié ici de modifier les options.